graph/appender/sidecars_check.go

```diff
@@ -44,33 +44,29 @@ func (a *SidecarsCheckAppender) applySidecarsChecks(trafficMap graph.TrafficMap,
 		}
 
 		// get the pods for the node, either by app+version labels, or workload deployment
-		var podLabels string
 		var err error
+		var pods *models.Pods
 		switch n.NodeType {
 		case graph.NodeTypeWorkload:
-			podLabels, err = a.getWorkloadLabels(n.Namespace, n.Workload, k8s)
-			if err != nil {
-				continue
+			workload, err := business.Workload.GetWorkload(n.Namespace, n.Workload, false)
+			if err == nil {
+				pods = &workload.Pods
 			}
 		case graph.NodeTypeApp:
-			podLabels = a.getAppLabels(appLabel, n.App, versionLabel, n.Version)
+			podLabels := a.getAppLabels(appLabel, n.App, versionLabel, n.Version)
+			pods, err = business.Workload.GetPods(n.Namespace, podLabels)
 		default:
 			continue
 		}
-		pods, err := k8s.GetPods(n.Namespace, podLabels)
 		checkError(err)
 
-		if len(pods.Items) == 0 {
-			log.Warningf("Sidecar check found no pods Checking sidecars node [%s] num pods [%v]", n.ID, len(pods.Items))
+		if pods == nil || (pods != nil && len(*pods) == 0) {
+			log.Warningf("Sidecar check found no pods Checking sidecars node [%s]", n.ID)
 		}
 
-		// check each pod for sidecar, stop and flag at first pod missing sidecar
-		for _, pod := range pods.Items {
-			checker := pods_checker.SidecarPresenceChecker{Pod: &pod}
-			if _, ok := checker.Check(); !ok {
-				n.Metadata["hasMissingSC"] = true
-				break
-			}
+		if !pods.HasIstioSideCar() {
+			n.Metadata["hasMissingSC"] = true
+			break
```

Comment:
This `break` looks suspicious... in previous code it was breaking the nested for-loop, now it's breaking the first level for-loop...

Reply:
@jotak +1, that break needs to be removed, it will prevent nodes from being checked.

Reply:
+1 good catch.